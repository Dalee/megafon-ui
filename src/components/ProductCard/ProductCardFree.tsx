import * as React from 'react';
import * as PropTypes from 'prop-types';
import './ProductCardFree.less';
import Header from '../Header/Header';
import Paragraph from '../Paragraph/Paragraph';
import { cnCreate } from '../../utils/cn';
import { IShowcaseChildren } from './ProductCardFeatures';

interface IProductCardFreeProps {
    /** Custom class name */
    className?: string;
    /** Title */
    title?: string;
    /** Showcase params childrens list */
    params: Array<Partial<IShowcaseChildren>>;
}

const cn = cnCreate('product-card-free');
class ProductCardFree extends React.Component<IProductCardFreeProps, {}> {
    static propTypes = {
        className: PropTypes.string,
        params: PropTypes.array.isRequired,
        title: PropTypes.string,
    };

    static defaultProps: IProductCardFreeProps = {
        params: [],
    };

    render() {
        const { params, title, className }: IProductCardFreeProps = this.props;

        return (
            <div className={cn('', {}, className)}>
                {title && <Header className={cn('title')} as="h3" margin={false}>{title}</Header>}
                <ul className={cn('list')}>
                    {params.map((param: IShowcaseChildren): React.ReactNode =>
                        <li className={cn('item', { icon: !!param.svgIcon })} key={param.title}>
                            {param.svgIcon}
                            <Header className={cn('item-title')} as="h5" margin={false}>
                                {param.title}
                            </Header>
                            <Paragraph
                                className={cn('item-text')}
                                marginAll="none"
                                sizeAll="small">
                                {param.caption}
                            </Paragraph>
                        </li>
                    )}
                </ul>
            </div>
        );
    }
}

export default ProductCardFree;
