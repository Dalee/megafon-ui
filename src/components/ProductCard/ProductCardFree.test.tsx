import * as React from 'react';
import { shallow } from 'enzyme';
import ProductCardFree from './ProductCardFree';
import Checked from 'icons/checked_24.svg';

const props = {
    title: 'test',
    params: [
        {
            svgIcon: <Checked />,
            title: 'test',
            caption: 'test',
            value: 'test',
        },
    ],
};

describe('<ProductCardFree />', () => {
    it('it renders ProductCardFree', () => {
        const wrapper = shallow(<ProductCardFree {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
