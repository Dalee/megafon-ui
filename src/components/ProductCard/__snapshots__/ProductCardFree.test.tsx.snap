// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCardFree /> it renders ProductCardFree 1`] = `
<div
  className="product-card-free"
>
  <Header
    as="h3"
    className="product-card-free__title"
    color="black"
    margin={false}
  >
    test
  </Header>
  <ul
    className="product-card-free__list"
  >
    <li
      className="product-card-free__item product-card-free__item_icon"
      key="test"
    >
      <test-file-stub />
      <Header
        as="h5"
        className="product-card-free__item-title"
        color="black"
        margin={false}
      >
        test
      </Header>
      <Paragraph
        className="product-card-free__item-text"
        color="black"
        marginAll="none"
        sizeAll="small"
        weight="regular"
      >
        test
      </Paragraph>
    </li>
  </ul>
</div>
`;
